volmdlr/core.py

```diff
@@ -45,6 +45,181 @@
+    elif math.isclose(sol, -1, abs_tol=1e-6):
+        inner_angle = math.pi
+    else:
+        inner_angle = math.acos(sol)
+
+    if cross < 0:
+        return inner_angle
+
+    return volmdlr.TWO_PI - inner_angle
+
+
+def vectors3d_angle(vector1, vector2):
+    """
+    Computes the angle between two 3 dimensional vectors.
+
+    :param vector1: The fist 3 dimensional vector
+    :type vector1: :class:`volmdlr.Vector3D`
+    :param vector2: The second 3 dimensional vectors
+    :type vector2: :class:`volmdlr.Vector3D`
+    :return: The angle between the two vectors
+    :rtype: flaot
+    """
+    dot = vector1.dot(vector2)
+    theta = math.acos(dot / (vector1.norm() * vector2.norm()))
+    return theta
+
+
+def sin_cos_angle(u1, u2):
+    """
+    Returns an angle between 0 and 2*PI verifying cos(theta)=u1, sin(theta)=u2.
+
+    :param u1: The value of the cosinus of the returned angle
+    :type u1: float
+    :param u2: The value of the sinus of the returned angle
+    :type u2: flaot
+    :return: The angle verifying the two equations
+    :rtype: float
+    """
+    if u1 < -1:
+        u1 = -1
+    elif u1 > 1:
+        u1 = 1
+    if u2 < -1:
+        u2 = -1
+    elif u2 > 1:
+        u2 = 1
+
+    if u1 > 0:
+        if u2 >= 0:
+            theta = math.acos(u1)
+        else:
+            theta = volmdlr.TWO_PI + math.asin(u2)
+    else:
+        if u2 >= 0:
+            theta = math.acos(u1)
+        else:
+            theta = volmdlr.TWO_PI - math.acos(u1)
+    if math.isclose(theta, volmdlr.TWO_PI, abs_tol=1e-9):
+        return 0.
+    return theta
+
+
+def delete_double_pos(points, triangles):
+    points_to_indexes = {}
+
+    for index, point in enumerate(points):
+        if point not in points_to_indexes:
+            points_to_indexes[point] = [index]
+        else:
+            points_to_indexes[point].append(index)
+
+    new_points = []
+    index_to_modified_index = {}
+    for i, (point, indexes) in enumerate(points_to_indexes.items()):
+        new_points.append(point)
+        index_to_modified_index[indexes[0]] = i
+
+    index_to_new_index = {}
+
+    for indexes in points_to_indexes.values():
+        for index in indexes[1:]:
+            index_to_new_index[index] = indexes[0]
+
+    new_triangles = []
+    for face_triangles in triangles:
+        if face_triangles is None:
+            continue
+        new_face_triangles = []
+        for triangle_ in face_triangles:
+            new_triangle = []
+            for index in triangle_:
+                if index in index_to_new_index:
+                    modified_index = index_to_modified_index[
+                        index_to_new_index[index]]
+                else:
+                    modified_index = index_to_modified_index[index]
+                new_triangle.append(modified_index)
+            new_face_triangles.append(tuple(new_triangle))
+        new_triangles.append(new_face_triangles)
+
+    return new_points, new_triangles
+
+
```

Comment:
all these methods have been removed from core.py?

Reply:
I think they are now in geometry module